.github/workflows/ci.yml

```diff
@@ -0,0 +1,87 @@
+name: CI
+
+on:
+  push:
+    branches: [master, staging]
+  pull_request:
+    branches: [master, staging]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+env:
+  CHROME_BIN: /usr/bin/google-chrome
+
+permissions:
+  id-token: write
+  contents: write
+
+jobs:
+  test:
+    name: Checks & Build Test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 12
+      - name: Install Dependenices
+        run: npm install
+      - name: Run Test with CodeCov
+        env:
+          CHROME_BIN: /usr/bin/google-chrome
+        run: npm run test:codecov
+      - name: Run Lint & Prettier Tests
+        run: |
+          npm run lint:type
+          npm run prettier:check
+      - name: Build staging
+        run: npm run build:stage
+      - name: build production
+        run: npm run build:prod
+      - name: Code coverage test
+        if: success()
+        uses: codecov/codecov-action@v1.1.1
```

Comment:
is there any danger of the 2 build steps corrupting the codecov data from the test step?

maybe we should order the steps so we upload codecov data immediately after running the tests just in case. And because you probably want it to fail faster if the build or test fails I probably would just move the run tests step after the builds and immediately before codecov

Reply:
hmm, I guess we were running the steps in the current order already in travis because we were using the `after_success` stage

Reply:
I don't believe there is any danger in the build corrupting the codecov data since this is the order Travis had the steps as well.

I think you're right about getting it to fail faster by running the `lint` & `prettier` checks first., but we want it to test the application before building it

Reply:
I'm good with that, after we get this up and working we can split out this into separate parallel jobs if it feels too slow running this all in the same job. But that's a future optimization that we don't need to do here.

Reply:
Scratch that; it looks like it builds when it runs `test:codecov` so I'll move it below the builds.